app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java

```diff
@@ -64,8 +64,9 @@ public static String generateRandomId() {
     getCollectionReference(collection).document(key).delete();
   }
 
-  static void updateDocument(String key, Map<String, Object> updates, String collection) {
-    getCollectionReference(collection).document(key).update(updates);
+  static CompletableFuture updateDocument(String key, Map<String, Object> updates, String collection) {
```

Comment:
Is there reason why we changed update flow to return CompletableFutures?

Reply:
Because we need to deal with it in the UI. We need to know whether they were successful or not. 
getCollectionReference(collection).document(key).update(updates); returns a completable future. If there's an exception, it will not be caught.